src/DOT2WTDOTConvertor.sol

```diff
@@ -0,0 +1,152 @@
+
+    /// @inheritdoc ILSTConvert
+    function inputToken() external view override returns (address) {
+        return dot;
+    }
+
+    /// @inheritdoc ILSTConvert
+    function outputToken() external view override returns (address) {
+        return wtdot;
+    }
+
+    /// @inheritdoc ILSTConvert
+    function convert(uint256 inputAmount) external override returns (uint256) {
+        return _convert(inputAmount, msg.sender);
+    }
+
+    /// @inheritdoc ILSTConvert
+    function convertTo(
+        uint256 inputAmount,
+        address receiver
+    ) external override returns (uint256) {
+        require(
+            receiver != address(0),
+            "DOT2WTDOTConvertor: zero address not allowed"
+        );
+        return _convert(inputAmount, receiver);
+    }
+
+    /// @notice Convert `inputAmount` token and send output token to `receiver`.
+    /// @param inputAmount The input token amount to convert.
+    /// @param receiver The receiver for the converted output token.
+    /// @return outputAmount The output token amount.
+    function _convert(
+        uint256 inputAmount,
+        address receiver
+    ) internal returns (uint256 outputAmount) {
+        require(inputAmount != 0, "DOT2WTDOTConvertor: invalid input amount");
+        IERC20(dot).safeTransferFrom(msg.sender, address(this), inputAmount);
+
+        // params for tDOT pool fo StableAsset on Acala:
+        // tDOT pool id: 0
+        // assets length: 2
+        // asset index of DOT: 0
+        // asset index of LDOT: 1
+        // here deadcode these params
+        (bool valid, address[] memory assets) = IStableAsset(stableAsset)
+            .getStableAssetPoolTokens(0);
+        require(
+            valid && assets[0] == dot,
+            "DOT2WTDOTConvertor: invalid stable asset pool"
+        );
+        uint256[] memory paramAmounts = new uint256[](2);
+
+        if (inputAmount.div(2) >= HOMA_MINT_THRESHOLD) {
+            uint256 beforeLdotAmount = IERC20(ldot).balanceOf(address(this));
+            bool suc = IHoma(homa).mint(inputAmount.div(2));
+            require(suc, "DOT2WTDOTConvertor: homa mint failed");
+            uint256 afterLdotAmount = IERC20(ldot).balanceOf(address(this));
+            uint256 ldotAmount = afterLdotAmount.sub(beforeLdotAmount);
+
+            // convert LDOT amount to rebased LDOT amount as the param
+            // NOTE: the precision of Homa.getExchangeRate is 1e18
+            uint256 ldotParamAmount = ldotAmount
+                .mul(IHoma(homa).getExchangeRate())
+                .div(1e18);
+            paramAmounts[0] = inputAmount.sub(inputAmount.div(2));
+            paramAmounts[1] = ldotParamAmount;
+        } else {
+            paramAmounts[0] = inputAmount;
+            paramAmounts[1] = 0;
+        }
+
+        uint256 beforeTdotAmount = IERC20(tdot).balanceOf(address(this));
+        bool success = IStableAsset(stableAsset).stableAssetMint(
+            0,
+            paramAmounts,
+            0
```

Comment:
do we not care about slippage here? Is there any scenario that this might cause significant fund loss?

Reply:
we basically assume DOT/LDOT ratio is 1:1, which is not always the case but should be a close enough. it is currently unlikely the pool ratio is too far away from this to make a difference. even it does happen, add liquidity using the expected ratio shouldn't yield a lose after the ratio back to normal